python/tvm/relax/transform/legalize_ops/ccl.py

```diff
@@ -46,6 +46,30 @@ def _allreduce(_bb: BlockBuilder, call: Call) -> Expr:
     )
 
 
+@register_legalize("relax.ccl.allgather")
+def _allgather(_bb: BlockBuilder, call: Call) -> Expr:
+    output_shape = []
+    assert isinstance(
+        call.args[0].struct_info, TensorStructInfo
+    ), "The input struct info of allgather should be TensorStructInfo."
+    assert isinstance(call.args[0].struct_info.shape.struct_info, ShapeStructInfo)
+    arg_shape = call.args[0].struct_info.shape.struct_info
+    for i, shape_value in enumerate(arg_shape.values):
+        if i == 0:
+            output_shape.append(tir.multiply(shape_value, call.args[1].value))
```

Comment:
Nit: The explicit call to `tir.multiply` can be replaced by `shape_value * call.args[1].value`, using operator overloading to delegate to `PrimExpr.__mul__`.   (Whether this is good or bad practice is a topic of a great many flamewars, so I tend to stick with whatever an existing codebase uses.)

Reply:
That make sense. Resolved.